packages/core/test/model.test.ts

```diff
@@ -0,0 +1,46 @@
+import { createMachine } from '../src';
+import { createModel } from '../src/model';
+
+describe('createModel', () => {
+  it('model.machine creates a machine that is updated', () => {
+    type UserEvent = {
+      type: 'updateName';
+      value: string;
+    };
+
+    interface UserContext {
+      name: string;
+      age: number;
+    }
+
+    const userModel = createModel<UserContext, UserEvent>({
+      name: 'David',
+      age: 30
+    }).withUpdaters({
+      updateName: {
+        name: (_, e: UserEvent) => e.value
+      }
+    });
+
+    const machine = createMachine<typeof userModel['context'], UserEvent>({
+      context: userModel.context,
+      initial: 'active',
+      states: {
+        active: {
+          on: {
+            updateName: {
```

Comment:
There seems to be still quite a bit of a boilerplate to write using the current approach and from what I understand the main motivation for this (at least at the beginning) was lowering the barrier for newcomers. So the question is - does it deliver that?

Reply:
It will, but I want to save that for a follow-up PR.